src/workflow/ArcanistWorkflow.php

```diff
@@ -2068,5 +2100,74 @@ private function getModernCommonDictionary(array $map) {
     return $map;
   }
 
+  // UBER CODE
+  // usso itself is somewhat sluggish, takes 1 second to return cached token
+  const USSO_CACHE_TIMEOUT = 600;
+
+  private static function getUSSOCacheFilename($domain) {
+    return implode(DIRECTORY_SEPARATOR,
+                   array(
+                   sys_get_temp_dir(),
+                         sprintf('usso-token-cache-%s.json', md5($domain)),
+                   ));
+  }
+
+  private function maybeUseUSSOToken($domain) {
+    $cache = self::getUSSOCacheFilename($domain);
+    try {
+      $data = @json_decode(file_get_contents($cache), true);
```

Comment:
Hiding errors will make debugging difficult. Use only if necessary.

Reply:
It is shortcut to check if file is readable and json is available - even if I check them only thing I can do is skip, as we discussed writing message that cache is empty doesn't make much sense for end user, in worst case user will see ~1second delay due to slow usso response